## Viable Insurance Client Information 

In [1]:
import pandas as pd
viable = pd.read_csv("Viable_Insurance.csv",index_col=0)
viable.head(5)

Education EmploymentStatus  Income Marital Status  Coverage  \
id                                                                           
0                College         Disabled   20819        Married  Extended   
1               Bachelor         Employed   35624         Single     Basic   
2   High School or Below         Employed   32272        Married     Basic   
3                 Master         Employed   48015        Married  Extended   
4                College         Disabled   26156         Single     Basic   

   Gender Location Code  Monthly Premium Auto  Months Since Last Claim  \
id                                                                       
0       F         Rural                    72                       22   
1       F      Suburban                    61                       22   
2       F      Suburban                   104                        1   
3       F         Rural                    80                       34   
4       F         Urban                    72                       10   

    Months Since Policy Inception  Number of Open Complaints  \
id                                                             
0                              72                          2   
1                              37                          0   
2                              78                          0   
3                              82                          3   
4                              41                          0   

    Number of Policies Claim Reason Sales Channel  
id                                                 
0                    1         Hail        Branch  
1                    1         Hail           Web  
2                    7        Other   Call Center  
3                    1        Other   Call Center  
4                    1    Collision         Agent

# Client Backgrounds

## 1) Education 

In [2]:
education=viable["Education"]
education_count=education.value_counts()
education_count

College                 1523
High School or Below    1458
Bachelor                1443
Master                   396
Doctor                   180
Name: Education, dtype: int64

## 2) Exmployment

In [3]:
employment=viable["EmploymentStatus"]
employment_count=employment.value_counts()
employment_count

Employed         3124
Unemployed       1270
Disabled          241
Medical Leave     234
Retired           131
Name: EmploymentStatus, dtype: int64

## 3) Income

In [4]:
income=viable["Income"]
income.describe()

count     5000.000000
mean     37915.415800
std      30516.918542
min          0.000000
25%          0.000000
50%      34738.000000
75%      62531.750000
max      99981.000000
Name: Income, dtype: float64

In [5]:
viable.loc[income<=30000,"Income"]="low income"
viable.loc[(income>30000) & (income<=60000),"Income"]="middle income"
viable.loc[(income>60000),"Income"]="high income"
viable["Income"].value_counts()

low income       2281
high income      1389
middle income    1330
Name: Income, dtype: int64

## 4) Marital Status

In [6]:
marital_Status=viable["Marital Status"]
marital_Status.value_counts()

Married     2918
Single      1353
Divorced     729
Name: Marital Status, dtype: int64

## 5) Gender

In [7]:
gender=viable["Gender"]
gender.value_counts()

F    2567
M    2433
Name: Gender, dtype: int64

## 6) Location

In [8]:
location=viable["Location Code"]
location.value_counts()

Suburban    3186
Rural        966
Urban        848
Name: Location Code, dtype: int64

## Main Customer Group

In [9]:
main=viable["Education"]+ ", "+viable["EmploymentStatus"]+", "+viable["Income"]+", "+viable["Marital Status"]+", "+viable["Location Code"]
main.value_counts()[0:10]

High School or Below, Employed, high income, Married, Suburban      187
College, Employed, middle income, Married, Suburban                 182
Bachelor, Employed, high income, Married, Suburban                  182
College, Employed, high income, Married, Suburban                   175
High School or Below, Unemployed, low income, Single, Suburban      169
High School or Below, Employed, middle income, Married, Suburban    168
College, Unemployed, low income, Single, Suburban                   153
Bachelor, Employed, middle income, Married, Suburban                144
Bachelor, Unemployed, low income, Single, Suburban                  133
Bachelor, Unemployed, low income, Married, Suburban                  84
dtype: int64

## Conclusion

From the above analysis, we conclude that our main client target is post secodary school education or below, employed, married people living in suburban area.

# Insurance Covereage/Policy Info

## Coverage

In [10]:
coverage=viable["Coverage"]
coverage.value_counts()

Basic       3057
Extended    1500
Premium      443
Name: Coverage, dtype: int64

## Policy Info

In [11]:
monthly_premium=viable[["Monthly Premium Auto","Months Since Policy Inception","Number of Policies"]]
monthly_premium.describe()

Monthly Premium Auto  Months Since Policy Inception  Number of Policies
count           5000.000000                    5000.000000         5000.000000
mean              93.613800                      48.500400            2.933000
std               35.320506                      27.888362            2.392333
min               61.000000                       0.000000            1.000000
25%               68.000000                      24.000000            1.000000
50%               83.000000                      49.000000            2.000000
75%              109.000000                      72.000000            4.000000
max              298.000000                      99.000000            9.000000

In [12]:
inception=viable["Months Since Policy Inception"]
customers_new=viable[(inception<=12)]
shape_new=customers_new.shape
print("Number of customers below 1 year: " + str(shape_new[0]))
customers_middle=viable[(inception>12)&(inception<60)]
shape_middle=customers_middle.shape
print("Number of customers over 1 year and below 5 years: " + str(shape_middle[0]))
customers_old=viable[(inception>=60)]
shape_old=customers_old.shape
print("Number of customers over 5 year: " + str(shape_old[0]))

Number of customers below 1 year: 625
Number of customers over 1 year and below 5 years: 2453
Number of customers over 5 year: 1922


# Claims Info

## Claim frequency in the recent year

In [13]:
month_lastclaim=viable["Months Since Last Claim"]
month_lastclaim.describe()

count    5000.000000
mean       15.037200
std        10.081654
min         0.000000
25%         6.000000
50%        14.000000
75%        23.000000
max        35.000000
Name: Months Since Last Claim, dtype: float64

In [22]:
last_claims=viable[month_lastclaim<12]
number_of_claims=last_claims.shape[0]
print(number_of_claims)
claim_frequency=number_of_claims/5000
print("The claim frequency1 is "+ str(claim_frequency))
#last_claims_2=viable[(month_lastclaim>=12)&(month_lastclaim<24)]
#number_of_claims_2=last_claims_2.shape[0]
#print(number_of_claims_2)
#claim_frequency2=number_of_claims_2/5000
#print("The claim frequency2 is "+ str(claim_frequency2))
#last_claims3=viable[month_lastclaim>=24]
#number_of_claims3=last_claims3.shape[0]
#print(number_of_claims3)
#claim_frequency3=number_of_claims3/5000
#print("The claim frequency3 is "+ str(claim_frequency3))

2129
The claim frequency1 is 0.4258


## Claim Reason

In [15]:
reason=viable["Claim Reason"]
percentage=reason.value_counts()/5000*100
print(percentage)

Collision       40.90
Hail            32.40
Scratch/Dent    15.06
Other           11.64
Name: Claim Reason, dtype: float64


The main reason for filing a claim is Collision(40.9%), the second is Hail(32.4%), while Scartch/Dent and Other accounts for the rest

## Claim and Location

In [16]:
claim_col=viable.loc[reason=="Collision","Location Code"]
print("Collision distribution")
print(claim_col.value_counts())
claim_hail=viable.loc[reason=="Hail","Location Code"]
print("\n"+ "Hail distribution")
print(claim_hail.value_counts())

Collision distribution
Suburban    1280
Rural        406
Urban        359
Name: Location Code, dtype: int64

Hail distribution
Suburban    1035
Rural        309
Urban        276
Name: Location Code, dtype: int64


Most claims happens in Suburban area, so we can consider raising the premium when we have a client from suburban area

# Customer Service/ Sales Channel

## Complaints

In [17]:
complaints=viable["Number of Open Complaints"]
number_complaints=complaints.sum()
print("The number of complaints is",number_complaints)
print("The average complaints per client ", number_complaints/5000)

The number of complaints is 1991
The average complaints per client  0.3982


The number of complaints of the company is too high. According to research by [Esteban Kolsky](https://www.slideshare.net/ekolsky/cx-for-executives), 13% of unhappy customers will share their complaint with 15 or more people. So settling open complaints is significant to keep our current customers as well as attracting new customers.

## Sales Channel

In [18]:
sales=viable["Sales Channel"]
sales.value_counts()

Agent          1903
Branch         1437
Call Center     961
Web             699
Name: Sales Channel, dtype: int64

Majority of our customers are introduced to our company through agents. As Covid-19 has a huge impact on people's life, marketing in-person has become less welcomed. Also, this can be costly as we need to hire more people in order for faster expansion. I recommend that we should invest in web development and online advertisemnet. 